nxc/protocols/smb.py

```diff
@@ -224,7 +224,7 @@ def enum_host_info(self):
             if not self.targetDomain:   # Not sure if that can even happen but now we are safe
                 self.targetDomain = self.hostname
         else:
-            self.hostname = self.host
+            self.hostname = self.host.split(".")[0]
             self.targetDomain = self.hostname
```

Comment:
If you have no ntlm, passing dc.domain.local as target will not work.


This split only occur if you have in a not ntlm condition so the target is always a fqdn

Reply:
Hmm okay makes sense, but this will break the implementation with ip:
![image](https://github.com/Pennyw0rth/NetExec/assets/61382599/3a7ec0cf-6d1f-4699-ad9b-4943e28f6e75)

Gonna take a look at how to fix it

Reply:
After some testing it appears that a hostname and a domain name must be supplied for Kerberos auth if NTLM is not available (the latter normally provides hostname). For now these three combinations work are available without NTLM:
![image](https://github.com/Pennyw0rth/NetExec/assets/61382599/1519e9d4-b9a0-4dcb-b175-33b18aab3271)